hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStore.java

```diff
@@ -78,12 +79,34 @@ public class RDBStore implements DBStore {
    */
   private final String dbCompactionLogDirName = "compaction-log";
 
+  // this is to track the total size of dbUpdates data since sequence
+  // number in request to avoid increase in heap memory.
+  private long maxDbUpdatesSizeThreshold;
+
   @VisibleForTesting
   public RDBStore(File dbFile, ManagedDBOptions options,
                   Set<TableConfig> families) throws IOException {
     this(dbFile, options, new ManagedWriteOptions(), families,
         new CodecRegistry(), false, 1000, null, false,
         TimeUnit.DAYS.toMillis(1), TimeUnit.HOURS.toMillis(1));
+    this.maxDbUpdatesSizeThreshold = MAX_DB_UPDATES_SIZE_THRESHOLD;
+  }
+
+  @SuppressWarnings("checkstyle:ParameterNumber")
+  @VisibleForTesting
+  public RDBStore(File dbFile, ManagedDBOptions rocksDBOption,
+                  ManagedWriteOptions writeOptions,
+                  Set<TableConfig> tableConfigs, CodecRegistry registry,
+                  boolean openReadOnly, int maxFSSnapshots,
+                  String dbJmxBeanNameName, boolean enableCompactionLog,
+                  long maxTimeAllowedForSnapshotInDag,
+                  long pruneCompactionDagDaemonRunInterval,
+                  long maxDbUpdatesSizeThreshold) throws IOException {
+    this(dbFile, rocksDBOption, writeOptions, tableConfigs, registry,
+        openReadOnly, maxFSSnapshots, dbJmxBeanNameName,
+        enableCompactionLog, maxTimeAllowedForSnapshotInDag,
+        pruneCompactionDagDaemonRunInterval);
+    this.maxDbUpdatesSizeThreshold = maxDbUpdatesSizeThreshold;
   }
```

Comment:
Please add the new param to the existing constructor instead of creating a new one.

Reply:
@adoroszlai - I have handled this review comment. Pls re-review.